cartridge-cli.lua

```diff
@@ -472,33 +466,31 @@ local function detect_git_version(source_dir)
     return version, release
 end
 
-local function find_rockspec(source_dir)
-    local files, err = utils.listdir(source_dir)
-    if files == nil then return nil, err end
-
-    for _, file in ipairs(files) do
-        if string.endswith(file, '.rockspec') then
-            return file
-        end
+local function find_rockspec(dir)
+    if dir == nil then
+        return nil, format_internal_error('find_rockspec: dir should be not nil')
     end
-end
-
-local function detect_name(source_dir)
-    local rockspec_filename, err = find_rockspec(source_dir)
-    if rockspec_filename == nil then return nil, err end
 
-    local rockspec_filepath = fio.pathjoin(source_dir, rockspec_filename)
-    local rockspec, err = utils.load_variables_from_file(rockspec_filepath)
-    if rockspec == nil then
-        return nil, string.format('Failed to load rockspec %s: %s', rockspec_filepath, err)
+    local rockspecs = fio.glob(fio.pathjoin(dir, '*-scm-1.rockspec'))
+    if #rockspecs == 1 then
+        return rockspecs[1]
     end
 
-    local name = rockspec.package
-    if name == nil then
-        return nil, string.format("Rockspec %s doesn't contain required field 'package'", rockspec_filepath)
+    if #rockspecs > 1 then
+        return nil, string.format('Found multiple rockspecs in %s', dir)
```

Comment:
Could I specify directly rockspec/version that I need?

Reply:
You can specify appname explicitly. In fact, you have to do it, if appname can't be detected from rockspec.